src/protocols/secure_channel/CASESessionCache.cpp

```diff
@@ -0,0 +1,107 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <protocols/secure_channel/CASESessionCache.h>
+
+namespace chip {
+
+CASESessionCache::CASESessionCache() {}
+
+CASESessionCache::~CASESessionCache()
+{
+    mCachePool.ForEachActiveObject([&](auto * ec) {
+        mCachePool.ReleaseObject(ec);
+        return true;
+    });
+}
+
+CASESessionSerializable * CASESessionCache::GetLRUSession()
+{
+    uint64_t minTimeStamp                = UINT64_MAX;
+    CASESessionSerializable * lruSession = nullptr;
+    mCachePool.ForEachActiveObject([&](auto * ec) {
+        if (minTimeStamp > ec->mSessionSetupTimeStamp)
+        {
+            minTimeStamp = ec->mSessionSetupTimeStamp;
+            lruSession   = ec;
+        }
+        return true;
+    });
+    return lruSession;
+}
+
+CHIP_ERROR CASESessionCache::Add(CASESession & session)
```

Comment:
This should not cache CASESessions, it should cache SecureSession. CASESession is the bit for just processing the session establishment. Once the session is established, most of the costly CASESession state can be dropped.

Overall, we need to make this cache work against SecureSession established over CASE

Reply:
We have created a CASESessionCachableClass that contains only the bare minimum members needed to store the information that is pertinent to session resumption as per spec. This comment has been addressed. please re-review